db/db_impl.cc

```diff
@@ -718,6 +718,9 @@ Status DBImpl::FlushWAL(bool sync) {
     if (!s.ok()) {
       ROCKS_LOG_ERROR(immutable_db_options_.info_log, "WAL flush error %s",
                       s.ToString().c_str());
+      // In case there is a fs error we should set it globally to prevent the
+      // future writes
+      WriteStatusCheck(s);
```

Comment:
if `sync` is true can this eventually reach the `SyncWAL` below, which might return `Status::OK` even though `s` is not ok?

Reply:
Thanks for catching this. Yeah we should skip sync if s is not ok.

Reply:
@maysamyabandeh how about this comment?

Reply:
what about it?

Reply:
Sorry, I misread the change and thought the comment was not addressed.